win-bk390a.cpp

```diff
@@ -595,22 +714,22 @@ int WINAPI wWinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance, PWSTR lpCmdLin
 			do {
 				com_read_status = ReadFile(hComm, &temp_char, sizeof(temp_char), &bytes_read, NULL);
 				d[i] = temp_char;
+            i++;
```

Comment:
This needs to move down below the debug line, else the debug line will display a char that isn't yet populated with data.

*Generally* I prefer to leave the incrementing to the start or end of the loop so that you don't get caught out with middle-of-block quirks like this, but we can put it in here since it's a simple enough loop.

I need to also add error checking on the byte read count rather than assuming it's actually read a byte.  I'll do this after the PR is done rather than breaking things again. Not that it's very likely that it'll happen nor be critical.

Reply:
I gotcha. That's actually why I moved the i++ up though -- because the \n was causing an exit in the middle of the loop at a value of 10 even though 11 characters had been read. I guess it comes down to personal preference though, but I thought i==11 would be the more logical check. However this is your project and I will revert it if you wish.

Reply:
Yes, that's basically it, comes down to personal preferences; in the end one side of the code or the other ends up having to be adjusted to handle it.  I'll bring in the PR and make the adjustments.